packages/common/util.js

```diff
@@ -31,10 +31,13 @@ const log = require('./log');
  *
  * @alias module:util
  */
+const warned = new Set();
```

Comment:
Given the fact that Node will [create a closure around everything in this module](https://nodejs.org/api/modules.html#modules_the_module_wrapper), creating the `warn` Set object here works and keeps it isolated from the rest of the program. It _is_ still accessible by the other functions in this module.

I don't think it's worth holding up the review for, but one thing to consider might be creating an explicit closure for just the `warn` object and the function that actually uses it. Doing that will make it very clear where that `warned` object is intended to be used.

```js
exports.deprecate = (() => {
  const warned = new Set();

  return (name, version, alternative) => {
    if (warned.has(name)) return;

    let message = `${name} is deprecated after version ${version} and will be removed in a future release.`;
    if (alternative) message += ` Use ${alternative} instead.`;

    log.warn(message);
  };
})();
```